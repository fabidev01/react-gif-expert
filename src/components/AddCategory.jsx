import React, { useState } from 'react'
import PropTypes from "prop-types";

export const AddCategory = ({onAddCategory}) => {
  const [inputValue, setInputValue] = useState('');
  const onImputChange = ({target}) => { 
    setInputValue(target.value);
  };
  const onImputSubmit = (e) => {
    e.preventDefault();
    if((inputValue.trim()).length < 1) return;
    onAddCategory(inputValue.trim());
    setInputValue('');
  };
  return (
    <form aria-label='form' onSubmit={ onImputSubmit }>
      <input 
      type="text"
      placeholder='Buscar Gifs'
      onChange = {e => onImputChange(e) }
      value = { inputValue }
      />
    </form>
  )
}

AddCategory.propTypes = {
  onAddCategory: PropTypes.func.isRequired
}