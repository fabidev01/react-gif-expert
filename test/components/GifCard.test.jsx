import { render, screen } from "@testing-library/react"
import { GifCard } from "../../src/components/GifCard"

describe('pruebas en el gifCard', () => { 
  const title= 'titulo'
  const url= 'https://github.com/'
  test('Debe hacer match con el snapshot', () => {
    const { container } = render(<GifCard title={ title } url={ url }/>);
    expect(container).toMatchSnapshot();
  })
  test('Debe mostrar la imagen con el URL y el alt indicado', () => {
    render(<GifCard title={ title } url={ url } />);
    const {src, alt} = screen.getByRole('img');
    expect(src).toBe(url);
    expect(alt).toBe(title);
  });
  test('Debe mostrar el titulo en el componente', () => {
    render(<GifCard title={ title } url={ url } />);
    expect(screen.getByText(title)).toBeTruthy();
  });
})