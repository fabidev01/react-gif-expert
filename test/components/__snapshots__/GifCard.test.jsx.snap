// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en el gifCard Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="titulo"
      src="https://github.com/"
    />
    <p>
      titulo
    </p>
  </div>
</div>
`;
